packages/types/src/task.ts

```diff
@@ -0,0 +1,98 @@
+import { RooCodeEventName } from "./events.js"
+import { type ClineMessage, type BlockingAsk, type TokenUsage } from "./message.js"
+import { type ToolUsage, type ToolName } from "./tool.js"
+
+/**
+ * TaskProviderLike
+ */
+
+export interface TaskProviderState {
+	mode?: string
+}
+
+export interface TaskProviderLike {
```

Comment:
Lol, can we use this for https://github.com/Kilo-Org/kilocode/pull/1821?
or are they also working on it?